### PR TITLE
Add contextual nav button

### PR DESCRIPTION
**Pull request by thilan-tran**
<p align="center">
  
</p>

## Changes

- added a new component `HeaderButton` exported from `NavBarScreen`
- by default, acts as a global back button in the navbar
- can set custom button actions for specific routes
  - eg. for the `Map` screen, change the icon to a menu that opens the sidebar

### CODE REVIEW

File: src/containers/NavBarScreen.tsx [hunk 1]
```diff
@@ -1,13 +1,64 @@
...
+	const defaultConfig: ButtonConfig = {
+		action: () => {
+			if (navigation.canGoBack()) {
+				navigation.goBack();
+			}
+		},
+		icon: 'ios-arrow-back',
+	};
+
+	const customConfig: { [screen: string]: ButtonConfig } = {
```
Comment (zhang-edward) [inline src/containers/NavBarScreen.tsx:30]: Can we add the hamburger menu action for the QR screen too?

**Review by zhang-edward — CHANGES_REQUESTED** [review]
Also, I think the `Dashboard` screen has a back button on the lower left. Can we get rid of that too, now that we have this contextual action?

File: src/containers/NavBarScreen.tsx [hunk 2]
```diff
@@ -18,6 +19,8 @@ interface ButtonConfig {
 }
 
 export const HeaderButton = ({ navigation, screen }: Props) => {
+	const { dispatch } = useContext(Ctx);
```
Comment (zhang-edward) [inline src/containers/NavBarScreen.tsx:22]: Do we need this?

**Review by zhang-edward — APPROVED** [review]
One nitpick

File: src/containers/NavBarScreen.tsx (hunk 2 above)
Comment (thilan-tran) [reply src/containers/NavBarScreen.tsx:22]: Whoops, meant to remove that as well in the `Remove deprecated context dispatch call` commit.

**Pull request merged**